lib/rex/proto/smb/simpleclient.rb

```diff
@@ -133,7 +148,13 @@ def login_split_next_ntlm1(user, domain, hash_lm, hash_nt)
 
   def connect(share)
     ok = self.client.tree_connect(share)
-    tree_id = ok['Payload']['SMB'].v['TreeID']
+
+    if ok.respond_to?(:id)
+      tree_id = ok.id
+    else
+      tree_id = ok['Payload']['SMB'].v['TreeID']
```

Comment:
this needs to be updated since  `ok['Payload']['SMB']` can be nil

Reply:
actually, now that I think about it, do we need to keep this code around at all?